src/Skia/Avalonia.Skia/Text/SKTextLayout.cs

```diff
@@ -0,0 +1,1610 @@
+
+                        // ToDo: proper width calculation of clusters with diacritics
+                        if (width < measuredWidth)
+                        {
+                            width = measuredWidth;
+                        }
+                    }
+                }
+
+                var point = glyphOffsets[currentCluster];
+
+                var rect = new SKRect(point.X, point.Y, point.X + width, point.Y + height);
+
+                glyphClusters.Add(new SKGlyphCluster(currentPosition, length, rect));
+
+                currentCluster = nextCluster;
+            }
+
+            return glyphClusters;
+        }
+
+        /// <summary>
+        /// Creates a list of text runs. Each text run only consists of one combination of text properties.
+        /// </summary>
+        /// <param name="text">The text.</param>
+        /// <param name="startingIndex">Index of the starting.</param>
+        /// <param name="length">The length.</param>
+        /// <returns></returns>
+        private unsafe IReadOnlyList<SKTextRun> CreateTextRuns(
+            ReadOnlyMemory<char> text,
+            int startingIndex,
+            int length)
+        {
+            var textRuns = new List<SKTextRun>();
+            var textPosition = 0;
+
+            var runText = text.Length == length ? text : text.Slice(startingIndex, length);
+
+            while (textPosition < length)
+            {
+                int glyphCount;
+
+                using (var pinHandle = runText.Pin())
+                {
+                    glyphCount = Math.Min(
+                        runText.Length,
+                        _typeface.CountGlyphs((IntPtr)pinHandle.Pointer, runText.Length, SKEncoding.Utf16));
+                }
+
+                if (runText.Length > glyphCount)
+                {
+                    var c = runText.Span[glyphCount];
+
+                    if (IsBreakChar(c))
+                    {
+                        glyphCount++;
+
+                        if (glyphCount < runText.Length)
+                        {
+                            switch (c)
+                            {
+                                case '\r' when runText.Span[glyphCount] == '\n':
+                                case '\n' when runText.Span[glyphCount] == '\r':
+                                    glyphCount++;
+                                    break;
+                            }
+                        }
+                    }
+                }
+
+                var typeface = _typeface;
+
+                if (glyphCount == 0)
+                {
+                    var codePoint = char.IsHighSurrogate(runText.Span[0])
+                                        ? char.ConvertToUtf32(runText.Span[0], runText.Span[1])
+                                        : runText.Span[0];
+
+                    typeface = SKFontManager.Default.MatchCharacter(codePoint);
+
+                    if (codePoint > sizeof(short))
```

Comment:
You've probably meant short.MaxValue here

Reply:
Thanks